lib/api/v1/controllers/results.go

```diff
@@ -36,6 +37,29 @@ func (c *ResultsController) List(ctx *gin.Context) {
 	RenderJSON(ctx, http.StatusOK, response)
 }
 
+func (c *ResultsController) Show(ctx *gin.Context) {
+	if c.EnsureAuthenticatedUser(ctx) != nil {
+		return
+	}
+
+	resultID, err := strconv.Atoi(ctx.Param("id"))
+	if err != nil {
+		RenderJSONError(ctx, errors.ErrInvalidRequest, err.Error())
+
+		return
+	}
+
+	result, err := models.GetResultByID(int64(resultID), []string{"User", "AdLinks", "Links"})
```

Comment:
Do you miss that part that the report is belonging to the current user? 🤔

Reply:
Nice catch! fixed in [bb1d257](https://github.com/carryall/go-google-scraper-challenge/pull/105/commits/bb1d257b202a0d940afbb0f6676a6bea743fd9be)